org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/lsp/WorkspaceFoldersProposedClient.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.lsp;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+
+public interface WorkspaceFoldersProposedClient {
+
+	/**
+	 * The workspace/workspaceFolders request is sent from the server to the client
```

Comment:
why is it never called by the server in this PR?

Reply:
We get the initial set in initialize, and from then on just listen to deltas.